Sources/PrimerSDK/Classes/Core/Payment Services/VaultService.swift

```diff
@@ -21,23 +21,24 @@ internal class VaultService: VaultServiceProtocol {
         }
         
         let api: PrimerAPIClientProtocol = DependencyContainer.resolve()
+        
+        firstly {
+            api.vaultFetchPaymentMethods(clientToken: clientToken)
+        }
+        .done { paymentMethods in
+            state.paymentMethods = paymentMethods.data
 
-        api.vaultFetchPaymentMethods(clientToken: clientToken) { [weak self] (result) in
```

Comment:
Refactoring this method is more risky since vault is a popular use case for mobile. Not against this, just want to make sure we're thoroughly testing this before we go ahead.